### ISSUE TITLE
Sockets can't be moved

### ISSUE THREAD

**Issue by schveiguy** [MartinNowak/io#18]
With dmd 2.083.0, I can't move a `Socket` using `std.algorithm.move`. I think this is a compiler bug, but I'm not sure why this fails.

I can call `Socket.move` no problem, but I can't call `std.algorithm.move(Socket)`.

I can easily duplicate the problem in a small test case with:

```D
import std.algorithm : move;

struct S
{
    int x;
    S foo() return nothrow @nogc pure { return S(x); }
    this(this) @disable;
}

void main()
{
    S s(2);
    auto s2 = s.foo; // ok
    auto s3 = move(s); // errors
}
```

Here are the errors (I removed some trivial path info):
```
std/algorithm/mutation.d(1232): Error: struct `testmove.S` is not copyable because it is annotated with @disable
std/algorithm/mutation.d(1225): Error: template instance `std.algorithm.mutation.moveImpl!(S)` error instantiating
std/algorithm/mutation.d(1183):        instantiated from here: trustedMoveImpl!(S)
```
The really odd thing is that the a member function MUST BE THERE, and it MUST have `return` attribute on it.

I will file an equivalent bug in bugzilla with this test case.

**Comment by schveiguy** [comment]
Added bugzilla issue: https://issues.dlang.org/show_bug.cgi?id=19415

**Comment by MartinNowak** [comment]
Should be fixed with the upcoming [v2.084.0-beta.1](https://dlang.org/changelog/2.084.0.html).
